lib/Test/Pod/CoverageChange.pm

```diff
@@ -0,0 +1,250 @@
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+=item * C<$ignored_packages> - arrayref of packages that will be ignored in the checks (optional)
+
+=back
+
+=cut
+
+sub check_pod_syntax {
+    my $path             = shift;
+    my $ignored_packages = shift;
+    my $Test_Builder     = Test::More->builder;
+
+    my @ignored_packages_full_path = ();
+    for (@$ignored_packages) {
+        my $file_path = module_path($_);
+        push @ignored_packages_full_path, $file_path if defined $file_path;
+    }
+
+    my @files_path = File::Find::Rule->file()->name('*.p[m|l]')->in(@$path);
+
+    for my $file_path (@files_path) {
+        next if any { /\Q$file_path/ } @ignored_packages_full_path;
+
+        my $check_result = podchecker($file_path);
+        if ($check_result == POD_SYNTAX_IS_OK) {
+            $Test_Builder->ok(1, sprintf("Pod structure is OK in the file %s.", $file_path));
+        } elsif ($check_result == FILE_HAS_NO_POD) {
+            $Test_Builder->todo_skip(sprintf("There is no POD in the file %s.", $file_path));
+        } else {
+            $Test_Builder->ok(0, sprintf("There are %d errors in the POD structure in the %s.", $check_result, $file_path));
+        }
+    }
+}
+
+=head2 check_allowed_naked_packages
+
+Checks passed allowed_naked_packages against existing package files.
+
+=over 4
+
+=item * C<$allowed_naked_packages> - hashref of number of allowed naked subs, keyed by package name (optional)
+
+=item * C<$ignored_packages> - a list of packages that will be ignored in our checks, supports arrayref (optional)
+
+=back
+
+Possible results
+
+=over 4
+
+=item * B<todo fail> if the numbers of existing naked subs are equal to passed value.
+
+=item * B<fails> if the numbers of existing naked subs are more/less than the passed value.
+
+=item * B<fails> if a package has 100% POD coverage and it passed as a L<$allowed_naked_package>.
+
+=back
+
+=cut
+
+sub check_allowed_naked_packages {
+    my $allowed_naked_packages = shift;
+    my $ignored_packages       = shift;
+    my $caller_test_file       = (caller(2))[1];
+    my $Test_Builder           = Test::More->builder;
+
+    # Check for the currently naked packages POD.
+    foreach my $package (sort keys %$allowed_naked_packages) {
+        next if any { /^\Q$package\E$/ } @$ignored_packages;
+
+        my $pc = Pod::Coverage->new(
+            package => $package,
+            private => []);
+        my $fully_covered = defined $pc->coverage && $pc->coverage == 1;
+        my $coverage_percentage     = defined $pc->coverage ? $pc->coverage * 100 : 0;
+        my $max_expected_naked_subs = $allowed_naked_packages->{$package};
+        my $naked_subs_count        = scalar $pc->naked // scalar $pc->_get_syms($package);
+
+        if (!$fully_covered) {
+            $Test_Builder->todo_skip(sprintf("You have %.2f%% POD coverage for the module '%s'.", $coverage_percentage, $package));
```

Comment:
Is this really a todo_skip?

Reply:
Yes, Because it's a `foreach` on $allowed_naked_packages we expect the number of the naked packages but we don't want to fail the tests, so I put this on a `todo_skip`